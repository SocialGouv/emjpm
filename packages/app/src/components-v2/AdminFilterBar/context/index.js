export {
  Context as AdminFilterContext,
  Provider as AdminFilterProvider,
  Consumer as AdminFilterConsumer
} from "./FiltersContext";
