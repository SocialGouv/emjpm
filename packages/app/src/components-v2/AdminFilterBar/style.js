const AdminFilterBarStyle = {
  width: "100%"
};

const FilterTextStyle = {
  textTransform: "uppercase",
  fontFamily: "body",
  fontSize: "11px",
  fontWeight: "600",
  letterSpacing: ".76px",
  lineHeight: "44px",
  mr: 1
};

export { AdminFilterBarStyle, FilterTextStyle };
