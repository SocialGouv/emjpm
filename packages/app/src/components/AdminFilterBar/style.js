const AdminFilterBarStyle = {
  width: "100%",
};

const FilterTextStyle = {
  fontFamily: "body",
  fontSize: "11px",
  fontWeight: "600",
  letterSpacing: ".76px",
  lineHeight: "44px",
  mr: 1,
  textTransform: "uppercase",
};

export { AdminFilterBarStyle, FilterTextStyle };
