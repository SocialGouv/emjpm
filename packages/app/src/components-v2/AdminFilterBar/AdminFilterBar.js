import { Card, Input, Text } from "@socialgouv/emjpm-ui-core";
import React, { useContext } from "react";
import { Box, Flex } from "rebass";
import { AdminFilterContext } from "./context";
import { AdminFilterBarStyle, FilterTextStyle } from "./style";

const AdminFilterBar = () => {
  const { searchText, changeSearchText } = useContext(AdminFilterContext);
  return (
    <Card mt="3" sx={AdminFilterBarStyle} mb={2}>
      <Flex justifyContent={"space-between"} flexWrap="wrap">
        <Box>
          <Flex>
            <Text sx={FilterTextStyle}>AFFINER LES RÃ‰SULTATS</Text>
            <Box width="170px" mr={1}>
              <Input
                value={searchText}
                spellCheck="false"
                autoComplete="false"
                onChange={event => changeSearchText(event.target.value)}
                name="search"
                size="small"
                placeholder="Filtre"
              />
            </Box>
          </Flex>
        </Box>
      </Flex>
    </Card>
  );
};

export { AdminFilterBar };
