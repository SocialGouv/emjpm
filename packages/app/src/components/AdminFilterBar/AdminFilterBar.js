import { Button, Card, Input, Select, Text } from "@emjpm/ui";
import React, { useContext } from "react";
import { Box, Flex } from "rebass";

import { AdminFilterContext } from "./context";
import { AdminFilterBarStyle, FilterTextStyle } from "./style";

const TYPE_OPTIONS = [
  { label: "Tous les types", value: null },
  { label: "Mandataire individuel", value: "individuel" },
  { label: "Mandataire préposé d'établissement", value: "prepose" },
  { label: "Membre d'un service", value: "service" },
  { label: "Magistrat", value: "ti" },
  { label: "Direction", value: "direction" },
  { label: "Admin", value: "admin" },
];

const AdminFilterBar = ({ onAddButtonClick, userTypeFilter }) => {
  const { searchText, changeSearchText, selectedType, selectType } = useContext(AdminFilterContext);
  return (
    <Card mt="3" sx={AdminFilterBarStyle} mb={2}>
      <Flex justifyContent={"space-between"} flexWrap="wrap">
        <Box>
          <Flex>
            <Text sx={FilterTextStyle}>AFFINER LES RÉSULTATS</Text>
            <Box width="170px" mr={1}>
              <Input
                value={searchText}
                spellCheck="false"
                autoComplete="false"
                onChange={(event) => changeSearchText(event.target.value)}
                name="search"
                size="small"
                placeholder="Filtre"
              />
            </Box>
            {userTypeFilter && (
              <Box width="170px" mr={1}>
                <Select
                  value={TYPE_OPTIONS.find((elm) => elm.value === selectedType)}
                  options={TYPE_OPTIONS}
                  onChange={(option) => selectType(option.value)}
                  name="type"
                  size="small"
                  placeholder="Type"
                />
              </Box>
            )}
          </Flex>
        </Box>
        {onAddButtonClick && (
          <Box>
            <Button width="120px" onClick={onAddButtonClick}>
              Ajouter
            </Button>
          </Box>
        )}
      </Flex>
    </Card>
  );
};

export { AdminFilterBar };
