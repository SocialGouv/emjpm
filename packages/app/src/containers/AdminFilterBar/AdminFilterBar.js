import { useContext } from "react";
import { Box, Flex } from "rebass";

import useQueryReady from "~/hooks/useQueryReady";
import { Button, Card, Input, Select, Text } from "~/components";
import { useDepartementsOptions } from "~/utils/departements";
import { findOption } from "~/utils/form";

import { AdminFilterContext } from "./context";
import { AdminFilterBarStyle, FilterTextStyle } from "./style";

const TYPE_OPTIONS = [
  { label: "Tous les types", value: null },
  { label: "Mandataire individuel", value: "individuel" },
  { label: "Mandataire préposé d'établissement", value: "prepose" },
  { label: "Membre d'un service", value: "service" },
  { label: "Magistrat", value: "ti" },
  { label: "Direction", value: "direction" },
  { label: "Admin", value: "admin" },
];

const departementsOptionsConfig = {
  nullOption: {
    label: "Tous les départements",
  },
};

function AdminFilterBar({
  onAddButtonClick,
  userTypeFilter,
  useDepartementfilter,
}) {
  const { departementsOptions, error, loading } = useDepartementsOptions(
    departementsOptionsConfig
  );

  const {
    searchText,
    changeSearchText,
    selectedType,
    selectType,
    selectedDepartementCode,
    selectDepartementCode,
  } = useContext(AdminFilterContext);

  if (!useQueryReady(loading, error)) {
    return null;
  }

  return (
    <Card mt="3" sx={AdminFilterBarStyle} mb={2}>
      <Flex justifyContent={"space-between"} flexWrap="wrap">
        <Box>
          <Flex>
            <Text sx={FilterTextStyle}>AFFINER LES RÉSULTATS</Text>
            <Box width="170px" mr={1}>
              <Input
                value={searchText}
                spellCheck="false"
                autoComplete="false"
                onChange={(event) => changeSearchText(event.target.value)}
                name="search"
                size="small"
                placeholder="Filtre"
              />
            </Box>
            {userTypeFilter && (
              <Box width="170px" mr={1}>
                <Select
                  instanceId={"user-type-filter"}
                  value={TYPE_OPTIONS.find((elm) => elm.value === selectedType)}
                  options={TYPE_OPTIONS}
                  onChange={(option) => selectType(option.value)}
                  name="type"
                  size="small"
                  placeholder="Type"
                />
              </Box>
            )}
            {useDepartementfilter && (
              <Box width="250px" mr={1}>
                <Select
                  instanceId={"departement-filter"}
                  value={findOption(
                    departementsOptions,
                    selectedDepartementCode
                  )}
                  options={departementsOptions}
                  onChange={(option) => selectDepartementCode(option.value)}
                  name="departement"
                  size="small"
                  placeholder="Département"
                />
              </Box>
            )}
          </Flex>
        </Box>
        {onAddButtonClick && (
          <Box>
            <Button width="120px" onClick={onAddButtonClick}>
              Ajouter
            </Button>
          </Box>
        )}
      </Flex>
    </Card>
  );
}

export { AdminFilterBar };
