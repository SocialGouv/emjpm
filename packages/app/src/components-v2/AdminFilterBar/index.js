import { AdminFilterBar } from "./AdminFilterBar";

export { AdminFilterBar };
