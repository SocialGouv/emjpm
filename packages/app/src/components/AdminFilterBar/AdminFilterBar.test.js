import React from "react";
import renderer from "react-test-renderer";

import { AdminFilterBar } from "./AdminFilterBar";

it("renders correctly", () => {
  const tree = renderer.create(<AdminFilterBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
