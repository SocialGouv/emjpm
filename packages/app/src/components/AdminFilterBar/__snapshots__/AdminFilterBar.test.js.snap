// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1scvnbg"
>
  <div
    className="css-1on3m70"
  >
    <div
      className="css-vurnku"
    >
      <div
        className="css-4cffwv"
      >
        <div
          className="css-z5wunp"
        >
          AFFINER LES RÃ‰SULTATS
        </div>
        <div
          className="css-19d2gif"
        >
          <div
            className="css-gw0z0n"
          >
            <input
              aria-label="search"
              aria-required={false}
              autoComplete="false"
              className="css-1noia2n"
              name="search"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Filtre"
              required={false}
              spellCheck="false"
              value={null}
            />
            <label
              aria-label="search"
              className="css-95ca3t"
              htmlFor="search"
            >
              Filtre
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-vurnku"
    >
      <button
        className="css-1k95xnl"
      >
        <div
          className="css-o6z23g"
        >
          Ajouter
        </div>
      </button>
    </div>
  </div>
</div>
`;
